src/topology/generic_pipeline.rs

```diff
@@ -29,12 +32,81 @@ pub struct Pipeline<T> {
     batch_config: BatchConfig,
     processors: Vec<String>,
     flush_listener: Option<FlushReceiver>,
+    resource_attributes: Vec<KeyValue>,
 }
 
 pub trait Inspect<T> {
     fn inspect(&self, value: &[T]);
 }
 
+pub trait ResourceAttributeSettable {
+    fn set_or_append_attributes(&mut self, attributes: Vec<KeyValue>);
+}
+
+impl ResourceAttributeSettable for ResourceSpans {
+    fn set_or_append_attributes(&mut self, attributes: Vec<KeyValue>) {
+        let mut drop_count = 0;
+        if self.resource.is_some() {
+            for rs in self.resource.iter() {
+                drop_count = rs.dropped_attributes_count;
+            }
```

Comment:
Isn't there always only one resource, I'm confused by the iter here, is it to unwrap the resource?

Could this whole function block be wrapped in a `if let Some(res) = self.resource { ... }` alternatively?